### DISCUSSION TITLE
How to toggle contextual code search  ?

### DISCUSSION THREAD

**Discussion by dtrckd** [TabbyML/tabby#3306] [Q&A]
After indexing one or more repositories with tabby (context provider), is it possible to:
1. Enable or disable code search functionality with the tabby-agent?
2. Specify certain paths or file extensions to limit which repositories should be included in the code search results used for the completion?

**Comment by wsxiaoys** [comment]
I assume you're referring to the homepage search box, also known as the Answer Engine. It is already possible to use it for contextual code search, which is only triggered when using the `#` symbol to select a relevant code repository.

<img width="1078" alt="image" src="https://github.com/user-attachments/assets/82aea3a6-6d8a-43df-8b8a-b23637094992">

**Reply by dtrckd** [reply]
Hey, thanks for your answer. 
No, I have re-edited my question which was badly written.
I am referring to the use of indexed code repositories chunks potentially used for the code completion (additional context given to the llm). But maybe it is not used like that and I am missing something ?

**Reply by wsxiaoys** [reply]
The repository code context is automatically used in the completion request. Please check https://github.com/TabbyML/tabby/issues/2674 for more details on the priority in the current implementation.
